analyzer/src/main/resources/scripts/init.gradle

```diff
@@ -230,6 +230,24 @@ class AbstractDependencyTreePlugin<T> implements Plugin<T> {
          */
         private Dependency fetchDependency(DependencyResult dependencyResult, Project project,
                                            Set<ResolvedArtifact> resolvedArtifacts, Set<String> visited) {
+            // Ignore this dependency if it is a BOM imported from Maven, because BOMs do not define dependencies but
+            // version constraints.
+            // See: https://docs.gradle.org/current/userguide/platforms.html#sub:bom_import
+            if (dependencyResult instanceof ResolvedDependencyResult &&
+                    dependencyResult.selected.hasProperty('variants')) {
+                def isBom = dependencyResult.selected.variants.every {
```

Comment:
Maybe extract the logic to check for a BOM into a separate function for better readability.